packages/backend/src/open_payments/quote/service.ts

```diff
@@ -0,0 +1,365 @@
+import assert from 'assert'
+import axios from 'axios'
+import { createHmac } from 'crypto'
+import { ModelObject, TransactionOrKnex } from 'objection'
+import * as Pay from '@interledger/pay'
+
+import { Pagination } from '../../shared/baseModel'
+import { BaseService } from '../../shared/baseService'
+import { QuoteError, isQuoteError } from './errors'
+import { Quote } from './model'
+import { Amount } from '../amount'
+import { AccountService } from '../account/service'
+import { RatesService } from '../../rates/service'
+import { IlpPlugin, IlpPluginOptions } from '../../shared/ilp_plugin'
+
+const MAX_INT64 = BigInt('9223372036854775807')
+
+export interface QuoteService {
+  get(id: string): Promise<Quote | undefined>
+  create(options: CreateQuoteOptions): Promise<Quote | QuoteError>
+  getAccountPage(accountId: string, pagination?: Pagination): Promise<Quote[]>
+}
+
+export interface ServiceDependencies extends BaseService {
+  knex: TransactionOrKnex
+  slippage: number
+  quoteUrl: string
+  quoteLifespan: number // milliseconds
+  signatureSecret?: string
+  signatureVersion: number
+  accountService: AccountService
+  ratesService: RatesService
+  makeIlpPlugin: (options: IlpPluginOptions) => IlpPlugin
+}
+
+export async function createQuoteService(
+  deps_: ServiceDependencies
+): Promise<QuoteService> {
+  const deps = {
+    ...deps_,
+    logger: deps_.logger.child({ service: 'QuoteService' })
+  }
+  return {
+    get: (id) => getQuote(deps, id),
+    create: (options: CreateQuoteOptions) => createQuote(deps, options),
+    getAccountPage: (accountId, pagination) =>
+      getAccountPage(deps, accountId, pagination)
+  }
+}
+
+async function getQuote(
+  deps: ServiceDependencies,
+  id: string
+): Promise<Quote | undefined> {
+  return Quote.query(deps.knex).findById(id).withGraphJoined('asset')
+}
+
+export interface CreateQuoteOptions {
+  accountId: string
+  sendAmount?: Amount
+  receiveAmount?: Amount
+  receivingAccount?: string
+  receivingPayment?: string
+}
+
+async function createQuote(
```

Comment:
This function is huge. Do you think it is worth it breaking it up a bit?

Reply:
https://github.com/interledger/rafiki/pull/326/commits/1ac83c18d4e207df50dcc98ad5c61db84fef8812